python/tank/commands/update.py

```diff
@@ -660,11 +667,29 @@ def _process_item(
         d["new_descriptor"] = status["latest"]
         d["app_name"] = app_name
         d["engine_name"] = engine_name
+        d["framework_name"] = framework_name
         d["env_name"] = env
 
         updated_items.append(d)
+
+        # Add the new descriptor to the descriptor lookup so that if
+        # we come across another instance of the new descriptor we can skip it.
+        # We don't add a key based on the original descriptor since there might
+        # be multiple instances of the old descriptor that need updating.
+        desc_key = self._get_descriptor_dict_lookup(status["latest"])
+        self._descriptor_look_up[desc_key] = status
+
         return updated_items
 
+    def _get_descriptor_dict_lookup(self, descriptor):
+        """
+        Converts the descriptors dictionary into an immutable value that
+        can be used as a key in a dictionary.
+        :param descriptor: Descriptor
+        :return: frozenset
+        """
+        return frozenset(descriptor.get_dict().items())
```

Comment:
It may be possible that I should just use `descriptor.get_uri()` but I wanted to be as closer to the raw descriptor descriptor as entered into the YAML file, and avoid the processing of the URI incase it made things that should be different the same.